src/Generator/Template/MethodTemplate.php

```diff
@@ -11,16 +12,18 @@ class MethodTemplate implements TemplateInterface
 {
     use VisibilityTrait;
 
-    const TEMPLATE = '
+    private const TEMPLATE = '
         %s %s function %s(%s)%s
         {
             %s $this->__call("%s", func_get_args());
         }
     ';
 
     /**
-     * @param \ReflectionMethod $method
+     * @param \Reflector|\ReflectionMethod $method
```

Comment:
Same as above.

Reply:
Because we call a specific method of the concrete class implementation.

Reply:
That's why the docblock had \ReflectionMethod instead of \Reflector. I mean that there shouldn't be any \Reflector type hint in the docblock.

Reply:
According to @Giuffre it's a matter of IDE warnings... Resolving.